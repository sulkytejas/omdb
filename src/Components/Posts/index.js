import React,{Component} from 'react';
import PropTypes from 'prop-types';
import SinglePost from './singlePost';
import { ButtonFilter } from '../../Utils/misc';

import {getFirstTitle,getRestofTitles,getfilterType} from '../../Utils/api';

class Posts extends Component {

  state = {
    post: [],
    page: 1,
    filter:false,
    type:"",
  }

  constructor(props) {
    super(props);

    //infinte scroll
    window.onscroll = () => {
      const {page,filter,type} = this.state
      if (
       window.innerHeight + document.documentElement.scrollTop
       === document.documentElement.offsetHeight
      ) {
        if (filter) {
          this.onScrollApi(type,page + 1,getfilterType);
        }else{
          this.onScrollApi(page,type,getRestofTitles);
        }

     }
    }
  }

  //infinte scroll helper function
  onScrollApi = (type,page,api) =>{
    const {post} = this.state
    api(type,page).then(a=>{
      if(a.Response && post.length <= a.totalResults){
       const allPosts = post.concat(a.Search)
       const increment = page + 1;
       this.setState({post:allPosts,page:increment})
      }
    })
  }

  //loading first 10 titles on page load
  componentDidMount(){
    getFirstTitle().then(a=> {
      this.setState({post:a.Search})
    })
  }

  //click event for filters
  handleClick = (type) => {
    const page = 1
    getfilterType(type,page).then(res=> {
      this.setState({post:res.Search,page,type,filter:true,})
    })
  }

  render() {
    const {post} = this.state
    const filterTypes = ['movie','game','series'];
    return (
      <div
        style={{
          display:'flex',
          flexDirection: 'column'
        }}
          >
        <div
          style=
          {{
            margin:20,
            alignSelf: 'center'
          }}
          >
          {filterTypes.map((a,i) => (
            <ButtonFilter  key={i} type={a} onClick={() => (this.handleClick(a))} >{a}</ButtonFilter>
          )
        )}
        </div>

        {post.map((a,i) => (
          <SinglePost
            key = {i}
            image= {a.Poster}
            name = {a.Title}
            date = {a.Year}
            />
        ))}

      </div>
    );
  }
}

Posts.propTypes = {
  post:PropTypes.array,
  page:PropTypes.number,
  filter:PropTypes.bool,
  type:PropTypes.string,

}

export default Posts;
