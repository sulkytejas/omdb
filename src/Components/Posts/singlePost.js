import React from 'react'
import PropTypes from 'prop-types'
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import CardMedia from '@material-ui/core/CardMedia';
import CardActionArea from '@material-ui/core/CardActionArea';
import ErrorImage from '../../Utils/images/image-error.png'


const SinglePost = (props) => {
  return (
    <Card
      style={{
        maxWidth:345,
        margin:20,
        alignSelf: 'center'}
      }>
      <CardActionArea>
        <CardMedia
            style ={{height:140}}
            image={props.image === 'N/A' ? ErrorImage : props.image }
            title="Contemplative Reptile"
          />
          <CardContent>
            <Typography
              gutterBottom
              variant="h5"
              component="h2">
              {props.name}
            </Typography>

            <Typography>
              {props.date}
            </Typography>
          </CardContent>
      </CardActionArea>
    </Card>
  )
}

SinglePost.propTypes = {
  image:PropTypes.string,
  name:PropTypes.string,
  date:PropTypes.string,
}

export default SinglePost
