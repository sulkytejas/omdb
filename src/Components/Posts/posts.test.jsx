import React from 'react';
import Posts from './index.js';
import {shallow,configure} from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('Posts',() =>{
  it('Check if posts state is set', () => {
    const wrapper = shallow(
      <Posts/>
    );
    wrapper.setState({post:[1,2,3]})
    expect(wrapper.state('post')).toEqual([1,2,3])
  });
})
