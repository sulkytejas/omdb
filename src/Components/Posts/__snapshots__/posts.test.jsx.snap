// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Posts render correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "alignSelf": "center",
        "margin": 20,
      }
    }
  >
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-outlined-9 MuiButton-outlinedPrimary-10"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "margin": 5,
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        movie
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-outlined-9 MuiButton-outlinedPrimary-10"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "margin": 5,
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        game
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-outlined-9 MuiButton-outlinedPrimary-10"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "margin": 5,
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        series
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
  </div>
</div>
`;
